scripts/lib/CIME/XML/env_batch.py

```diff
@@ -149,16 +144,16 @@ def set_batch_system(self, batchobj, batch_system_type=None):
             self.set_batch_system_type(batch_system_type)
 
         if batchobj.batch_system_node is not None and batchobj.machine_node is not None:
-            for node in batchobj.get_nodes('any', root=batchobj.machine_node, xpath='*'):
-                oldnode = batchobj.get_optional_node(node.tag, root=batchobj.batch_system_node)
-                if oldnode is not None and oldnode.tag != "directives":
-                    logger.debug( "Replacing {}".format(oldnode.tag))
-                    batchobj.batch_system_node.remove(oldnode)
+            for node in batchobj.get_children(root=batchobj.machine_node, no_validate=True):
```

Comment:
The '*' xpath was only needed because we didn't support a None nodename as an argument to get_nodes.